source/Halibut/ServiceModel/ServiceInvoker.cs

```diff
@@ -103,19 +207,28 @@ static MethodInfo SelectMethod(IList<MethodInfo> methods, RequestMessage request
             throw new AmbiguousMethodMatchHalibutClientException(message.ToString(), new AmbiguousMatchException(message.ToString()));
         }
 
+        async Task<object> InvokeAsyncMethod(MethodInfo method, object obj, params object[] parameters)
+        {
+            var task = (Task)method.Invoke(obj, parameters);
+            await task.ConfigureAwait(false);
+            var resultProperty = task.GetType().GetProperty("Result");
```

Comment:
Is it possible to avoid the reflection?

Reply:
No, I don't think so. ServiceInvoker is already primarly based on reflection, I'm not sure how to do it another way.